Repository/PhpcrOdmRepository.php

```diff
@@ -116,4 +126,87 @@ protected function buildCollection(array $documents)
 
         return $collection;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add($path, $resource)
+    {
+        Assert::startsWith($path, '/', 'Target path "%s" must be absolute.');
+
+        $resolvedPath = $this->resolvePath($path);
+        $parentDocument = $this->getManager()->find(null, $resolvedPath);
+
+        if (null === $parentDocument) {
+            throw new InvalidArgumentException(sprintf('Cannot locate parent document at "%s"', $resolvedPath));
```

Comment:
see `PhpcrOdmRepositoryTest::testAddResourceNotMatchingParentPath()`
not matching the message but the exception itslf.